sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -690,7 +690,15 @@ object PushPredicateThroughAggregate extends Rule[LogicalPlan] with PredicateHel
   def apply(plan: LogicalPlan): LogicalPlan = plan transform {
     case filter @ Filter(condition,
         aggregate @ Aggregate(groupingExpressions, aggregateExpressions, grandChild)) =>
-      val (pushDown, stayUp) = splitConjunctivePredicates(condition).partition {
+
+      // Create a map of Alias for grouping keys or literals
+      val aliasMap = AttributeMap(aggregateExpressions.collect {
+        case a: Alias if groupingExpressions.contains(a.child) || a.child.foldable =>
+          (a.toAttribute, a.child)
+      })
+      val newCond = PushPredicateThroughProject.replaceAlias(condition, aliasMap)
+
+      val (pushDown, stayUp) = splitConjunctivePredicates(newCond).partition {
         conjunct => conjunct.references subsetOf AttributeSet(groupingExpressions)
```

Comment:
Can you see if you can make AttributeSet deal with this. It is more general and more consistent with what I understand the intent of AttributeSet is.

The comment in AttributeSet is clear that it wants AttributeReferences which Aliases are not. Is that possible?